src/code.cloudfoundry.org/cni-wrapper-plugin/netrules/rule_converter.go

```diff
@@ -59,6 +59,22 @@ func (c *RuleConverter) BulkConvert(ruleSpec []Rule, logChainName string, global
 	return iptablesRules
 }
 
+func (c *RuleConverter) DeduplicateRules(iptablesRules []rules.IPTablesRule) []rules.IPTablesRule {
+	keys := make(map[string]bool)
+	deduped_rules := []rules.IPTablesRule{}
+
+	for _, rule := range iptablesRules {
+		key := strings.Join(rule, " ")
```

Comment:
What is the reason for this Join? Isn't 'rule' a single line string without multiple elements?

Reply:
Not exactly. A rule is an array of strings. Here we join all the elements into one string so it is easier to compare.